integration/spark/shared/src/test/java/io/openlineage/spark/agent/lifecycle/plan/JdbcRelationHandlerTest.java

```diff
@@ -91,6 +92,12 @@ void testHandlingJdbcTable() {
 
   @Test
   void testHandlingJdbcDbTableAsSubQuery() {
```

Comment:
Would it make sense to create a test `testHandlingJdbcDbTableAsSubQueryWithUnknownDialect`? I am not following how `testUnknownDialect` verifies a behaviour described within issue description.

Reply:
`testUnknownDialect` is just checking that if here `dialect` is `null`, query still can be parsed using generic dialect:
https://github.com/OpenLineage/OpenLineage/blob/62276dc99da5f51a112b9877c9fd8fe545db8fcc/integration/spark/shared/src/main/java/io/openlineage/spark/agent/util/JdbcSparkUtils.java#L105

Reply:
Redarding `testHandlingJdbcDbTableAsSubQueryWithUnknownDialect` test, I don't think it will make much sense. Both cases `table is not set, query=SELECT ...` and `table=(SELECT...)` are now handler in the same way, and covered by tests above

Reply:
Common practice about fixing issues and unit tests is to: (1) write a failing unit test and (2) fix a code to make a failing test pass. 

Issue description is 
```
openlineage-spark returns datasets with names like database.(select * from table where filter='value') T, not database.table
```
Is there any test verifying this? 
How does the introduced test `testUnknownDialect` relate to issue description and code that has been modified?

Reply:
> (1) write a failing unit test
> (2) fix a code to make a failing test pass

I've updated `testHandlingJdbcDbTableAsSubQuery` test to cover this. Previously it covered just the same case as `testHandlingJdbcQuery`